pkg/pillar/zedcloud/tls.go

```diff
@@ -172,6 +184,111 @@ func GetTlsConfig(dns *types.DeviceNetworkStatus, serverName string, clientCert
 	return tlsConfig, nil
 }
 
+func cacheProxyCerts(dns *types.DeviceNetworkStatus) [][]byte {
+	var certPEM [][]byte
+	// find all unique certs and save them
+	for _, port := range dns.Ports {
+		for _, pem := range port.ProxyCertPEM {
+			if !containsCert(certPEM, pem) {
+				certPEM = append(certPEM, pem)
+			}
+		}
+	}
+	return certPEM
+}
+
+func containsCert(certPEM [][]byte, pem []byte) bool {
+	for _, prevPEM := range certPEM {
+		if bytes.Equal(prevPEM, pem) {
+			return true
+		}
+	}
+	return false
+}
+
+// check to see if all the dns new proxy certs we already have
+func checkAllCerts(ctx *ZedCloudContext, dns *types.DeviceNetworkStatus) bool {
```

Comment:
Nit but the function name is not clear. diffProxyCerts() or checkProxyCertsChanged() would be much more clear.

Reply:
Sure. Updated